pkg/kubelet/eviction/eviction_manager.go

```diff
@@ -259,6 +260,17 @@ func (m *managerImpl) synchronize(diskInfoProvider DiskInfoProvider, podFunc Act
 		}
 		m.dedicatedImageFs = &hasImageFs
 		splitContainerImageFs := m.containerGC.IsContainerFsSeparateFromImageFs(ctx)
```

Comment:
do we even really need this check? if !m.dedicatedImageFs, then aren't we sure to not be split?

Reply:
We need to support the case where containerfs and imagefs are on the same disk but different from nodefs.

Reply:
I see, the function tests that? `IsContainerFsSeparateFromImageFs` seems like it's testing if imagefs==containerfs not if imagefs+containerfs == nodefs you know?

Reply:
yea. It looks to see if paths are different by talking with the CRI. It uses `ImageFsInfo` from container runtime to understand where these layers are stored. Since we only have path locations it will at least tell us if they are in different locations. 

The CRI doesn't really know about nodefs in this case. It only knows what the container runtime communicates back.

Reply:
yeah I guess my point in the original comment: would it really be a splitContainerImageFs if cadvisor didn't verify they're on different disks? ultimately that'd be the check. I don't see a situation where this containerGC check would help notice the contianerfs/imagefs are different from nodefs (don't we need to also ask cadvisor that?)

Reply:
True. I can rename the function. Its not possible to determine if a system is split from the just the CRI.

Reply:
CAdvisor is not exposed in the runtime service. It has access to imageService and runtimeService.

Reply:
We met offline. I’m going to drop this function call and introduce a dedicatedContainerFs function.

Reply:
PTAL @haircommander